CefSharp.BrowserSubprocess.Core/CefAppUnmanagedWrapper.cpp

```diff
@@ -385,8 +400,18 @@ namespace CefSharp
         }
         else if (name == kJavascriptRootObjectRequest)
         {
-            _javascriptAsyncRootObject = DeserializeJsRootObject(argList, 0);
-            _javascriptRootObject = DeserializeJsRootObject(argList, 1);
+            auto asyncRoot = DeserializeJsRootObject(argList, 0);
+            auto syncRoot = DeserializeJsRootObject(argList, 1);
+            _javascriptAsyncRootObjects->default[browser->GetIdentifier()] = asyncRoot;
+            _javascriptRootObjects->default[browser->GetIdentifier()] = syncRoot;
+
+            //let's store the non-popup browser's root objects to have default values for the popups
+            if (!browser->IsPopup())
```

Comment:
Would it be cleaner to move the assignment for the popup objects into the branch of the `if` statement then do a simple `HasKey` check above? Remove the need for https://github.com/cefsharp/CefSharp/pull/1562/files#diff-a27ca20dcc4f76e2dde0c785dd2abb9cR50

Overall I think we can make the intent a little cleaner to follow. Open to other suggestions